data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/producer/KafkaCustomProducerFactory.java

```diff
@@ -53,11 +54,23 @@ public KafkaCustomProducer createProducer(final KafkaProducerConfig kafkaProduce
                                               final boolean topicNameInMetrics) {
         AwsContext awsContext = new AwsContext(kafkaProducerConfig, awsCredentialsSupplier);
         KeyFactory keyFactory = new KeyFactory(awsContext);
-        prepareTopicAndSchema(kafkaProducerConfig);
+        // If either or both of Producer's max_request_size or
+        // Topic's max_message_bytes is set, then maximum of the
+        // two is set for both. If neither is set, then defaults are used.
+        Integer maxRequestSize = null;
+        KafkaProducerProperties producerProperties = kafkaProducerConfig.getKafkaProducerProperties();
+        if (producerProperties != null) {
+            int producerMaxRequestSize = producerProperties.getMaxRequestSize();
+            if (producerMaxRequestSize != KafkaProducerProperties.DEFAULT_MAX_REQUEST_SIZE) {
```

Comment:
What do you think about the following change?

```
if (producerMaxRequestSize > KafkaProducerProperties.DEFAULT_MAX_REQUEST_SIZE)
```

I think we could be ok with the default when the value defined is lower than the default.